lesson_7/edmond_sabou/lesson7_2.py

```diff
@@ -0,0 +1,25 @@
+def safe_divide(func):
+
+    def verify(*args):
+
+        try:
+            result = func(*args)
+        except ZeroDivisionError as zero_err:
+            print(f"Sorry! {zero_err} occurred. Try NOT dividing by zero!")
+            pass
+        except TypeError as type_err:
+            print(f"Oops! {type_err} error occurred.")
+            pass
+        else:
+            return f"The division result is: {result}"
+
+    return verify
+
+
+@safe_divide
+def divide(first_number, second_number):
+    return first_number / second_number
+
+
+print(divide(6, "w"))
+
```

Comment:
What would be the result if I try the following:
```python
a = divide(10, 2)
print(a)
```
?

Reply:
Is this a trick question?  the result would be 5. The code would still work.

Reply:
It's not a trick question and I was not asking what the result would theoretically be, but rather, if you run your code, what would the result be.
Please test

Reply:
did maybe mean to say ```print(a * 5)``` and not just `print(a)`?

Reply:
made a few changes to the code, so that it can handle the situation you proposed. thanks.

Reply:
Yap, it's better now, but I will like to ask you to make some additional changes; see the newly added inline comments

Reply:
made the change to `raise Exception.` I think I understand why this is better: because of the exit code, which is now 1, signaling that an actual error was encountered. until now even if an error has occurred the 0 exit code was printed.